### PR TITLE
TST: Add format, transformer, and type tests to FeatureTable

### LINKED ISSUE #43: add tests for feature_table subpackage
See `distance_matrix` subpackage tests (added in #40) for a guide on how to write and structure these tests.

### PR DESCRIPTION
**Pull request by jakereps**
Resolves #43

### CODE REVIEW

**Comment by jakereps** [comment]
Tests might pass or fail, depending on the dict randomness until qiime2/qiime2/pull/155 is merged.

File: q2_types/feature_table/tests/test_format.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+from q2_types.feature_table import (BIOMV100Format, BIOMV210Format,
+                                    BIOMV100DirFmt, BIOMV210DirFmt)
+from qiime.plugin.testing import TestPluginBase
+
+
+class TestFormats(TestPluginBase):
+    package = 'q2_types.feature_table.tests'
+
+    def test_biomv100_format_validate_positive(self):
+        filepath = self.get_data_path('feature-table_v100.biom')
```
Comment (thermokarst) [inline q2_types/feature_table/tests/test_format.py]: Can you swap `feature-table_v100.biom` out for something smaller?

```{"id": "None","format": "Biological Observation Matrix 1.0.0","format_url": "http://biom-format.org","type": "OTU table","generated_by": "QIIME 1.6.0-dev","date": "2013-02-09T09:30:11.550590","matrix_type": "sparse","matrix_element_type": "int","shape": [14, 9],"data": [[0,0,20],[0,1,18],[0,2,18],[0,3,22],[0,4,4],[1,4,1],[2,0,1],[2,4,1],[2,5,1],[3,6,1],[4,4,1],[5,7,20],[6,4,1],[7,4,1],[7,5,1],[8,4,1],[8,6,2],[8,8,3],[9,7,2],[10,5,1],[11,4,9],[11,5,20],[11,6,1],[11,8,4],[12,4,3],[12,6,19],[12,8,15],[13,0,1],[13,1,4],[13,2,4]],"rows": [{"id": "295053", "metadata": {"taxonomy": ["k__Bacteria"]}},{"id": "42684", "metadata": {"taxonomy": ["k__Bacteria", "p__Proteobacteria"]}},{"id": "None11", "metadata": {"taxonomy": ["Unclassified"]}},{"id": "None10", "metadata": {"taxonomy": ["Unclassified"]}},{"id": "None7", "metadata": {"taxonomy": ["Unclassified"]}},{"id": "None6", "metadata": {"taxonomy": ["Unclassified"]}},{"id": "None5", "metadata": {"taxonomy": ["k__Bacteria"]}},{"id": "None4", "metadata": {"taxonomy": ["Unclassified"]}},{"id": "None3", "metadata": {"taxonomy": ["k__Bacteria"]}},{"id": "None2", "metadata": {"taxonomy": ["k__Bacteria"]}},{"id": "None1", "metadata": {"taxonomy": ["Unclassified"]}},{"id": "879972", "metadata": {"taxonomy": ["k__Bacteria"]}},{"id": "None9", "metadata": {"taxonomy": ["Unclassified"]}},{"id": "None8", "metadata": {"taxonomy": ["k__Bacteria"]}}],"columns": [{"id": "f2", "metadata": null},{"id": "f1", "metadata": null},{"id": "f3", "metadata": null},{"id": "f4", "metadata": null},{"id": "p2", "metadata": null},{"id": "p1", "metadata": null},{"id": "t1", "metadata": null},{"id": "not16S.1", "metadata": null},{"id": "t2", "metadata": null}]}```

File: q2_types/feature_table/tests/test_transformer.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+        obs = transformer(input)
+        obs = biom.load_table(str(obs))
+
+        exp = input
+        self.assertEqual(obs.ids(axis='observation').all(),
+                         exp.ids(axis='observation').all())
+        self.assertEqual(obs.ids(axis='sample').all(),
+                         exp.ids(axis='sample').all())
+
```
Comment (thermokarst) [inline q2_types/feature_table/tests/test_transformer.py]: Two more transformers were added to this subpackage, can you add tests for those, too?

- [ ] `biom.Table -> pd.DataFrame`
- [ ] `BIOMV100Format -> BIOMV210Format`

Comment (jakereps) [inline q2_types/feature_table/tests/test_transformer.py]: Sounds good 👍

File: q2_types/feature_table/tests/test_format.py (hunk 1 above)
Comment (jakereps) [inline q2_types/feature_table/tests/test_format.py]: Swapped!

File: q2_types/feature_table/tests/test_transformer.py (hunk 2 above)
Comment (jakereps) [inline q2_types/feature_table/tests/test_transformer.py]: Done!

**Comment by jakereps** [comment]
Thanks for the review, @thermokarst! This should be good to go after tests 👍 !

**Pull request merged**